provisioning/WFGP-292_doc_galleon-pack.adoc

```diff
@@ -0,0 +1,164 @@
+* More generally, the tool that is used to generate this model reference (wildscribe) is lagging behind new WildFly management features (stability level was added recently while it's been in WildFly for some time)
+
+We can improve the user experience for WildFly users by consolidating our tooling and providing consistent documentation for the WildFly ecosystem.
+
+Each feature pack should generate user documentation that describes the management capabilities offered by the feature pack in a user-friendly readable output.
+This user documentation should be made available in https://docs.wildfly.org/ or the web site that documents other feature packs.
+
+This documentation should be generated when the feature pack is built and made available as a companion to the feature pack. This should not required additional work from the feature pack developer other than providing correct information about their feature packs.
+
+=== User Stories
+
+* As a WildFly user, I want to have documentation of the capabilities and resources that are provided by any feature packs from WildFly community.
+* As a feature pack developer, I want to provide documentation for my feature pack that is automatically generated whenever I build my feature pack.
+
+== Issue Metadata
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-292[WFGP-292 - Generate documentation for Feature Pack]
+
+=== Affected Projects or Components
+
+* WildFly Galleon Plugins - https://github.com/wildfly/galleon-plugins
+** Generate documentation when the `build-feature-pack` goal is invoked. It generates a companion doc archive next to the feature pack itself
+* Galleon - https://github.com/wildfly/galleon
+** Might be impacted if we are missing any metadata from the feature pack to generate its documentation
+* Wildscribe - https://github.com/wildfly/wildscribe
+** This RFE would be a full replacement of Wildscribe that would be archived.
+* Any project publishing feature packs (wildfly, grpc-feature-pack, ai-feature-pack, etc.)
+** They will be updated to bump the version of the Galleon maven plugin
+** *No other changes should be required*
+* WildFly - https://github.com/wildfly/
+** Its model references will be generated when its feature packs are built.
+*** For WildFly, we could generate a single documentation entry points that would document the wildfly-galleon-pack and also the capabilites coming from the wildfly-ee-galleon-pack.
+** The dependency to wildscribe from its docs module will be removed and replaced by unarchiving the documentation when the  wildfly-galleon-pack feature pack is built.
+** There would be no changes to the WildFly release process.
+
```

Comment:
Suggested detail requirement

* Documentation for feature packs that don't provide layers or don't provide their own management resources will be appropriately formatted.

I just intend this as a reminder to check this, or at least consciously handle it in the implementation so we don't end up with unprofessional output in these scenarios.

For example, wildfly-datasources-galleon-pack provides no new management resource API. https://github.com/wildfly-extras/deployment-transformer-feature-pack provides neither layers nor any new management API